src/runtime-rs/crates/shimv2/src/panic_hook.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{boxed::Box, ops::Deref};
+
+use backtrace::Backtrace;
+
+pub(crate) fn set_panic_hook() {
+    std::panic::set_hook(Box::new(move |panic_info| {
+        let (filename, line) = panic_info
+            .location()
+            .map(|loc| (loc.file(), loc.line()))
+            .unwrap_or(("<unknown>", 0));
+
+        let cause = panic_info
+            .payload()
+            .downcast_ref::<String>()
+            .map(std::string::String::deref);
+
+        let cause = cause.unwrap_or_else(|| {
+            panic_info
+                .payload()
+                .downcast_ref::<&str>()
+                .copied()
+                .unwrap_or("<cause unknown>")
+        });
+        let bt = Backtrace::new();
```

Comment:
This reminds me, the Kata 1.x runtime had a [`SIGUSR1` handler](https://github.com/kata-containers/runtime/blob/master/cli/main.go#L219..L222) that would log a formatted backtrace on receiving that signal. It could be useful to re-add that feature.

Reply:
Good idea, I'll try it, tokio may not be the same